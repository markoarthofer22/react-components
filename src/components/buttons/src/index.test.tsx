import React from 'react';
import TestRenderer from 'react-test-renderer';
import GlobalThemeProvider from '../../../themes/global-theme-wrapper/src/index';

import { Button } from './index';

describe('Button', () => {
    test('snapshot renders', () => {
        const component = TestRenderer.create(
            <GlobalThemeProvider>
                <Button
                    clicked={() => console.log('clicked')}
                    title='New title'
                />
            </GlobalThemeProvider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
