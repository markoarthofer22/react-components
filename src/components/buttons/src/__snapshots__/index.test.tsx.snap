// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot renders 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "html{line-height:1.15;-webkit-text-size-adjust:100%;}body{margin:0;}main{display:block;}h1{font-size:2em;margin:0.67em 0;}hr{box-sizing:content-box;height:0;overflow:visible;}pre{font-family:monospace,monospace;font-size:1em;}a{background-color:transparent;}abbr[title]{border-bottom:none;-webkit-text-decoration:underline;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted;}b,strong{font-weight:bolder;}code,kbd,samp{font-family:monospace,monospace;font-size:1em;}small{font-size:80%;}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline;}sub{bottom:-0.25em;}sup{top:-0.5em;}img{border-style:none;}button,input,optgroup,select,textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}button,input{overflow:visible;}button,select{text-transform:none;}button,[type=\\"button\\"],[type=\\"reset\\"],[type=\\"submit\\"]{-webkit-appearance:button;}button::-moz-focus-inner,[type=\\"button\\"]::-moz-focus-inner,[type=\\"reset\\"]::-moz-focus-inner,[type=\\"submit\\"]::-moz-focus-inner{border-style:none;padding:0;}button:-moz-focusring,[type=\\"button\\"]:-moz-focusring,[type=\\"reset\\"]:-moz-focusring,[type=\\"submit\\"]:-moz-focusring{outline:1px dotted ButtonText;}fieldset{padding:0.35em 0.75em 0.625em;}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal;}progress{vertical-align:baseline;}textarea{overflow:auto;}[type=\\"checkbox\\"],[type=\\"radio\\"]{box-sizing:border-box;padding:0;}[type=\\"number\\"]::-webkit-inner-spin-button,[type=\\"number\\"]::-webkit-outer-spin-button{height:auto;}[type=\\"search\\"]{-webkit-appearance:textfield;outline-offset:-2px;}[type=\\"search\\"]::-webkit-search-decoration{-webkit-appearance:none;}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit;}details{display:block;}summary{display:-webkit-box;display:-webkit-list-item;display:-ms-list-itembox;display:list-item;}template{display:none;}[hidden]{display:none;}@fontface{font-family:'Poppins',sans-serif;src:url('https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;700&family=Roboto:wght@400;500;700&display=swap');}@fontface{font-family:'Exo 2',sans-serif;src:url('https://fonts.googleapis.com/css?family=Exo+2:400,900,800,700,600,500,300,200,100');}html{font-family:'Poppins',sans-serif;width:100%;height:100%;box-sizing:border-box;-ms-overflow-style:scrollbar;}body{font-family:'Poppins',sans-serif;font-size:14px;line-height:1.231;height:100%;font-weight:500;color:#000000;background-color:#ffffff;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;overscroll-behavior-y:none;-webkit-font-smoothing:antialiased;position:static;overflow-y:auto;}p{font-size:14px;color:#000000;font-weight:500;line-height:16px;}h1,h2,h3,h4,h5,h6{margin:0.47em 0;font-weight:500;color:#000000;}h1{font-size:4rem;}h2{font-size:3.4rem;}h3{font-size:2rem;}h4{font-size:1.6rem;}h5{font-size:1.2rem;}h6{font-size:1.1rem;}b,strong{color:#000000;font-weight:700;}.mt-5{margin-top:5px;}.mb-5{margin-bottom:5px;}.mt-10{margin-top:10px;}.mb-10{margin-bottom:10px;}.mt-15{margin-top:15px;}.mb-15{margin-bottom:15px;}.mt-20{margin-top:20px;}.mb-20{margin-bottom:20px;}.mr-5{margin-right:5px;}.ml-5{margin-left:5px;}.mr-10{margin-right:10px;}.ml-10{margin-left:10px;}.mr-15{margin-right:15px;}.ml-15{margin-left:15px;}.mr-20{margin-right:20px;}.ml-20{margin-left:20px;}.popup-enter{opacity:0;-webkit-transform:scale(1);-moz-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}.popup-enter .window{-webkit-animation:ScaleIn 0.4s;animation:ScaleIn 0.4s;}.popup-enter-active{opacity:1;-webkit-transform:scale(1);-moz-transform:scale(1);-ms-transform:scale(1);transform:scale(1);-webkit-transition:opacity 0.4s,-webkit-transform 0.4s;transition:opacity 0.4s,transform 0.4s;}.popup-exit{opacity:1;}.popup-exit .window{-webkit-animation:ScaleIn 0.4s reverse;animation:ScaleIn 0.4s reverse;}.popup-exit-active{opacity:0;-webkit-transition:opacity 0.4s,-webkit-transform 0.4s;transition:opacity 0.4s,transform 0.4s;}@-webkit-keyframes scaleIn{0%{-webkit-transform:scale(0);-moz-transform:scale(0);-ms-transform:scale(0);transform:scale(0);}50%{-webkit-transform:scale(0.2);-moz-transform:scale(0.2);-ms-transform:scale(0.2);transform:scale(0.2);}100%{-webkit-transform:scale(1);-moz-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}}@keyframes scaleIn{0%{-webkit-transform:scale(0);-moz-transform:scale(0);-ms-transform:scale(0);transform:scale(0);}50%{-webkit-transform:scale(0.2);-moz-transform:scale(0.2);-ms-transform:scale(0.2);transform:scale(0.2);}100%{-webkit-transform:scale(1);-moz-transform:scale(1);-ms-transform:scale(1);transform:scale(1);}}.form-item-container{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.form-item-container:{-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.form-item-container: .form-item-floating{width:calc(100% - 20px);}.form-item-container: .form-item-floating:first-of-type{margin-right:20px;}.form-item-container: .form-item-floating.phone-type{width:calc(50% - 10px);margin-right:0;}.form-item-container.single .form-item-floating{width:100%;margin-right:0;}.form-item-container.radio-single{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;}.form-item-container.radio-single.inline:{-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}.form-item-floating{position:relative;margin-bottom:20px;}.form-item-floating--radio{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;height:40px;}.form-item-floating--checkbox{height:60px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;}.checkbox-single{margin-bottom:10px;position:relative;}.checkbox-single a{-webkit-transition:all 0.4s;transition:all 0.4s;color:#333;}.checkbox-single a:hover{color:#333;}.checkbox-single label{color:#333;}.checkbox-single span::before{left:35px;top:calc(100% - 10px);}",
      }
    }
    data-emotion="css-global 10rm74"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-4bgz60-Button{position:relative;padding:10px 25px;-webkit-transition:all 0.4s;transition:all 0.4s;cursor:pointer;display:inline-block;-webkit-transform:translateZ(0);-moz-transform:translateZ(0);-ms-transform:translateZ(0);transform:translateZ(0);border-radius:2px;text-align:center;background-color:transparent;border:1px solid transparent;font-weight:600;height:40px;font-size:14px;}.css-4bgz60-Button.disabled{pointer-events:none;opacity:0.5;}.css-4bgz60-Button.default{color:#ffffff;background-color:#000000;height:auto;}.css-4bgz60-Button.default:hover{color:#000000;background-color:#ffffff;border-color:#000000;-webkit-transform:scale(1.05);-moz-transform:scale(1.05);-ms-transform:scale(1.05);transform:scale(1.05);}.css-4bgz60-Button.red{color:#ffffff;background-color:#d71920;height:auto;}.css-4bgz60-Button.red:hover{color:#d71920;background-color:#ffffff;border-color:#d71920;-webkit-transform:scale(1.05);-moz-transform:scale(1.05);-ms-transform:scale(1.05);transform:scale(1.05);}.css-4bgz60-Button.pink{color:#ffffff;background-color:#c71585;height:auto;}.css-4bgz60-Button.pink:hover{color:#c71585;background-color:transparent;border-color:#c71585;-webkit-transform:scale(1.05);-moz-transform:scale(1.05);-ms-transform:scale(1.05);transform:scale(1.05);}.css-4bgz60-Button.green{color:#ffffff;background-color:#2e8b57;height:auto;}.css-4bgz60-Button.green:hover{color:#2e8b57;background-color:transparent;border-color:#2e8b57;-webkit-transform:scale(1.05);-moz-transform:scale(1.05);-ms-transform:scale(1.05);transform:scale(1.05);}.css-4bgz60-Button.blue{color:#ffffff;background-color:#1693c5;height:auto;}.css-4bgz60-Button.blue:hover{color:#1693c5;background-color:#ffffff;border-color:#1693c5;-webkit-transform:scale(1.05);-moz-transform:scale(1.05);-ms-transform:scale(1.05);transform:scale(1.05);}.css-4bgz60-Button.small{padding:5px 10px;}.css-4bgz60-Button.big{padding:12px 30px;}.css-4bgz60-Button.loading{color:transparent!important;background-color:#1693c5!important;opacity:1!important;}.css-4bgz60-Button.loading.clear{background-color:transparent!important;}.css-4bgz60-Button.loading::after{content:url(
                    'data:image/svg+xml;utf8,<svg version=\\"1.0\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 30 30\\" style=\\"enable-background:new 0 0 30 30;\\" xml:space=\\"preserve\\"><style type=\\"text/css\\">.st0{fill:none;stroke:url(%23XMLID_3_);stroke-width:2;stroke-linecap:round;stroke-miterlimit:10;}</style><linearGradient id=\\"XMLID_3_\\" gradientUnits=\\"userSpaceOnUse\\" x1=\\"3.9823\\" y1=\\"13.5647\\" x2=\\"26.0177\\" y2=\\"13.5647\\"><stop  offset=\\"0\\" style=\\"stop-color:%23d1af66\\"/><stop  offset=\\"1\\" style=\\"stop-color:%23d1af66\\"/></linearGradient><path id=\\"XMLID_2_\\" class=\\"st0\\" d=\\"M22.5,21.7c1.6-1.8,2.5-4.1,2.5-6.7c0-5.5-4.5-10-10-10S5,9.5,5,15c0,2.8,1.1,5.3,3,7.1\\"/></svg>' );position:absolute;opacity:1;left:50%;top:50%;margin-left:-15px;margin-top:-15px;width:30px;height:30px;background-color:transparent;color:#ffffff;-webkit-animation:animation-1ce3mz5 2s infinite linear;animation:animation-1ce3mz5 2s infinite linear;}@-webkit-keyframes animation-1ce3mz5{0%{-webkit-transform:rotate(0deg);-moz-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}100%{-webkit-transform:rotate(360deg);-moz-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}@keyframes animation-1ce3mz5{0%{-webkit-transform:rotate(0deg);-moz-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}100%{-webkit-transform:rotate(360deg);-moz-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}",
      }
    }
    data-emotion="css 4bgz60-Button animation-1ce3mz5"
  />,
  <button
    className="button   default  css-4bgz60-Button"
    disabled={false}
    onClick={[Function]}
    title="New title"
    type="button"
  >
    New title
  </button>,
]
`;
